pwndbg/dbg/lldb/repl/__init__.py

```diff
@@ -351,14 +351,29 @@ def handle_sigint(_sig, _frame):
 
         # At this point, the last command might've queued up some execution
         # control procedures for us to chew on. Run them now.
+        coroutine_fail_warn = False
         for process, coroutine in dbg.controllers:
             assert driver.has_process()
             assert driver.process.GetUniqueID() == process.process.GetUniqueID()
 
-            driver.run_coroutine(coroutine)
+            try:
+                driver.run_coroutine(coroutine)
+            except Exception:
+                # We treat exceptions coming from the execution controllers the
+                # same way we treat exceptions coming from commands.
+                pwndbg.exception.handle()
+                coroutine_fail_warn = True
```

Comment:
Should this break the `for process, coroutine in dbg.controllers:` loop?

Reply:
Well, I don't think there's a straight up right answer about what we should do there. On the one hand breaking there would lead to a single easier to read error, but on the other it will stop all the other accumulated ECs from executing, which might obfuscate other errors or leave things in an inconsistent state.

Currently, though, there is no case in Pwndbg where breaking there would make a difference, as no commands ever schedule more than one EC. But I think this is something we should revisit in the future, for sure, and by then it should be clearer what the best approach is.